lib/wnews.dart

```diff
@@ -1,48 +1,53 @@
+import 'dart:convert';
+
 import 'package:xpath_selector/xpath_selector.dart';
-import 'package:xpath_selector_html_parser/xpath_selector_html_parser.dart';
+import 'package:html/parser.dart' as html;
 
-/// Takes the HTML contents of something like
-/// https://en.wikipedia.org/wiki/Main_Page
-/// and extracts the items in the _In the news_ section.
+/// Parses the JSON response from
+/// https://api.wikimedia.org/feed/v1/wikipedia/en/featured/2023/10/30
+/// where date is current day and language is parameter
+/// and extracts news items.
 ///
-/// For the sake of simplicity, this currently only works for the English
-/// Wikipedia, and assumes quite a lot about the structure
-/// of the page.
-List<NewsItem> extractNews(String html) {
-  final parser = HtmlXPath.html(html);
+/// For each item uses story and links parameters
+List<NewsItem> extractNews(String json) {
+  final parsedJson = jsonDecode(json) as Map<String, dynamic>;
+  final news = parsedJson['news'] as List<dynamic>?;
 
-  final newsHeadline =
-      parser.query("//h2[contains(text(),'In the news')]").node;
-  if (newsHeadline == null) {
-    throw ArgumentError('Cannot find the headline "In the news"');
+  if (news == null || news.isEmpty) {
+    throw StateError('The list is empty');
   }
 
-  final boxDiv = newsHeadline.queryXPath('/ancestor::div').node;
-  if (boxDiv == null) {
-    throw StateError('Cannot find the ancestor <div> of "In the news"');
-  }
+  final newsList = news.map((n) {
```

Comment:
[nit] I generally prefer `for (...)` to functional mapping (`.map()`) when the body is this long, but in this case it's okay if you want to leave it this way.

Reply:
Updated